deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubSSLContext.java

```diff
@@ -280,16 +280,8 @@ private TrustManagerFactory generateTrustManagerFactory(IotHubCertificateManager
 
     private char[] generateTemporaryPassword()
     {
-        byte[] randomBytes = new byte[256];
-        char[] randomChars = new char[256];
-        new SecureRandom().nextBytes(randomBytes);
-
-        for (int i = 0; i < 256; i++)
-        {
-            randomChars[i] = (char) randomBytes[i];
-        }
-
-        return randomChars;
+        // removing all dash characters so that the password is alphanumeric
+        return UUID.randomUUID().toString().replace("-", "").toCharArray();
```

Comment:
A sample uuid format (and the corresponding error description it might generate) would help explain while the "-" removal is necessary